com.github.dnadolny.javatoscala/src/com/github/dnadolny/javatoscala/convertsnippet/PasteJavaAsScalaCommand.scala

```diff
@@ -12,6 +10,7 @@ import org.eclipse.swt.widgets.Display
 import org.eclipse.text.edits.ReplaceEdit
 import org.eclipse.ui.handlers.HandlerUtil
 import org.eclipse.ui.PlatformUI
+import org.scalaide.ui.internal.editor.ScalaSourceFileEditor
```

Comment:
As far as I can see you're only using methods from `ITextEditor`. If you only need to check that it's indeed a Scala editor, I think this is fine (you could change the pattern to `editor : ITextEditor if editor.isInstanceOf[ScalaSourceFileEditor]`. This way you wouldn't rely on any method that's not API.

There could be another Eclipse-blessed way to check if the "content-type" is Scala, but not sure how easy it is to find the content-type of a given editor